Controllers/CartController.cs

```diff
@@ -163,14 +160,69 @@ public IActionResult Index()
                 openOrder.DateCompleted = DateTime.Now;
                 context.Order.Update(openOrder);
                 await context.SaveChangesAsync();
-                return RedirectToAction("Index","Cart");
+                return RedirectToAction("Confirmation", new RouteValueDictionary(
+                     new { controller = "Cart", action = "Confirmation", Id = openOrder.OrderId } ));
 
             }
             catch (DbUpdateException)
             {
                 throw;
             }   
         }
+        public async Task<IActionResult> Confirmation(int id)
+        {
+            User user = ActiveUser.Instance.User;
```

Comment:
We don't need to grab the UserId here since we're getting the OrderId from the route. Line 174 - 180 can be deleted